### ISSUE TITLE
[tsserver] Custom Diagnostic errors, don't triggered getCodeFixAtPosition

### ISSUE THREAD

**Issue by sviat9440** [microsoft/TypeScript#35553]
When must be triggered getCodeFixAtPosition?
I think every time, when i open code fix menu in my IDE.

Note: I use InteljIDEA.

So, when I write code that has an internal (typescript) error (like syntax or type error), and open code fix menu (by ALT + ENTER as default), IDE call getCodeFixAtPosition with tsserver.
For example:
```
const a: number = ''; // type error: cant assign string to number
```
then getCodeFixAtPosition was called and works.

![image](https://user-images.githubusercontent.com/13107421/70369947-db9dba00-18d1-11ea-8fbc-aaad8dec607b.png)
![image](https://user-images.githubusercontent.com/13107421/70369946-d0e32500-18d1-11ea-8239-f98be65e631a.png)

But, when I create custom diagnostic error, in getSemanticDiagnostics or getSyntacticDiagnostics, then getCodeFixAtPosition was not called when I try to open code fix menu.

![image](https://user-images.githubusercontent.com/13107421/70369953-f2441100-18d1-11ea-80cc-b1e7b14b2f6f.png)
![image](https://user-images.githubusercontent.com/13107421/70369957-fa03b580-18d1-11ea-8d27-76e36ace0e91.png)

Here is code, that create custom error:
```
export function HooksChecker(base: LanguageServiceConstructor) {
    class HooksCheckerLanguageService extends base {
        private parser = new LanguageServiceParser(this.info.project);

        constructor(private info: ts.server.PluginCreateInfo) {
            super();
        }

        public getSemanticDiagnostics(fileName: string): ts.Diagnostic[] {
            return [
                ...super.getSemanticDiagnostics(fileName),
                {
                    category: DiagnosticCategory.Error,
                    code: 0,
                    file: this.parser.getSourceFile(fileName),
                    start: 0,
                    length: 5,
                    messageText: 'MyCustomError',
                },
            ];
        }

        public getCodeFixesAtPosition(
            fileName: string,
            start: number,
            end: number,
            errorCodes: ReadonlyArray<number>,
            formatOptions: ts.FormatCodeSettings,
            preferences: ts.UserPreferences,
        ): ReadonlyArray<ts.CodeFixAction> {
            return [
                ...super.getCodeFixesAtPosition(fileName, start, end, errorCodes, formatOptions, preferences),
                {
                    description: 'test test',
                    changes: [
                        {
                            fileName: fileName,
                            textChanges: [
                                {
                                    span: {
                                        start: 0,
                                        length: 2,
                                    },
                                    newText: 'test test',
                                }
                            ]
                        }
                    ],
                    fixName: 'test fix name'
                }
            ]
        }
    }
}
```

I don't know where to look for the problem. Maybe a bug in the IDE, but maybe I did something wrong.

**Comment by sviat9440** [comment]
Ok, I solved it. The problem was that I was sending error code = 0. But if you change it to any other, everything works.

**Comment by ArnaudBarre** [comment]
For anyone finding this, this not only need to be non 0, but this need to be an error code which is part of `getSupportedErrorCodes`.
Ideally a new error code can be added via `getSupportedCodeFixes`, but this is [not used by VSCode yet](https://github.com/microsoft/vscode/issues/219273)
Using the internal `codefix.registerCodeFix` works in VSCode, but [not in WebStorm](https://youtrack.jetbrains.com/issue/WEB-67884/getSupportedCodeFixes-is-called-before-TS-plugins-are-registered)
